data/vibe/data/serialization.d

```diff
@@ -285,6 +285,26 @@ unittest {
 	assert(test.text == "Hello");
 }
 
+unittest {
+	import vibe.data.json;
+	import std.exception;
+
+	enum Testable : string {
+		foo = "foo",
+		bar = "bar"
+	}
+
+	void deserializeString(string value) {
+		deserialize!(JsonSerializer, Testable)(Json(value));
+	}
+
+	foreach (string val; ["foo", "bar"]) {
+		assertNotThrown(deserializeString(val));
+	}
+
+	assertThrown!JSONException(deserializeString("foobar"));
+}
+
```

Comment:
Looks like the test should be in the `vibe.data.json` module.

Reply:
I suppose it should stay here if we decide to move the logic back into a `vibe.data.serialization` module ) Let's discuss below.

Reply:
I left it here since the serialization logic is also in the same module.